src/datalink/bpf.rs

```diff
@@ -159,10 +159,9 @@ pub struct DataLinkSenderImpl {
     header_size: usize,
 }
 
-impl DataLinkSenderImpl {
-    pub fn build_and_send<F>(&mut self, num_packets: usize, packet_size: usize,
-                          func: &mut F) -> Option<io::Result<()>>
-        where F : FnMut(MutableEthernetPacket)
+impl DataLinkSender for DataLinkSenderImpl {
+    fn build_and_send(&mut self, num_packets: usize, packet_size: usize,
```

Comment:
Could you mark all the `build_and_send` functions in the impls as `#[inline]` please? This should maximise the chance of the builder function getting inlined.